Fabulous.XamarinForms/tests/Fabulous.XamarinForms.Tests/ViewUpdaters/UpdateCollectionGenericTests.fs

```diff
@@ -7,113 +7,151 @@ open NUnit.Framework
 open FsUnit
 
 // 1 & 1' means its the same ViewElement (same property values), except it's not the same .NET reference
-// X & Y means its 2 different ViewElement types that can't be reused between themselves (e.g. Label vs Button)  
-module UpdateCollectionGenericTests =
-    let (-->) previous current = previous, current
-    let (?->) previous current = (ValueSome previous), current
-    
+// Tx & Ty means its 2 different ViewElement types that can't be reused between themselves (e.g. Label vs Button)
+// 1k/Txk means its a ViewElement with a key value 
+module UpdateCollectionGenericTests =    
     type Operation =
-        | Create of ViewElement
-        | Update of previous: ViewElement * current: ViewElement
-        | Remove of ViewElement
-        | Attach of previous: ViewElement voption * current: ViewElement
+        | Move of prevIndex: int * child: ViewElement
+        | Create of newChild: ViewElement
+        | Update of prevIndex: int * prevChild: ViewElement * newChild: ViewElement
+        | Remove of prevChild: ViewElement
+        | Clear
     
-    let private testUpdateCollectionGeneric previousCollection newCollection =
-        let operations = List<Operation>()
+    /// Call updateCollectionGenericInternal and accumulate all requested operations based on their index of effect
+    let private testUpdateCollectionGeneric (previousCollection: ViewElement list voption) (newCollection: ViewElement list voption) =
+        let operations = List<int * Operation>()
+        
+        let mockMove i child =
+            let prevIndex = previousCollection.Value |> List.findIndex (fun c -> c = child)
+            if(i<>prevIndex) then 
+             operations.Add(i, Move (prevIndex, child))
         
-        let mockCreate viewElement =
-            operations.Add(Create viewElement); viewElement.ToString() :> obj
+        let mockCreate i viewElement =
+            operations.Add(i, Create viewElement);
             
-        let mockUpdate previousViewElement currentViewElement target =
-            // When updating, the target control should always be in sync with the previous ViewElement
-            target |> should equal (previousViewElement.ToString() :> obj)
-            operations.Add(Update (previousViewElement, currentViewElement))
+        let mockUpdate i previousChild newChild =
+            let prevIndex = previousCollection.Value |> List.findIndex (fun c -> c = previousChild)
+            operations.Add(i, Update (prevIndex, previousChild, newChild))
             
-        let mockAttach previousViewElement currentViewElement _ =
-            operations.Add(Attach (previousViewElement, currentViewElement))
+        let mockRemove i unused=
+            let mutable offset = 0
+            for el in unused do
+             operations.Add(i + offset, Remove el)
+             offset<-offset+1
```

Comment:
Here the first parameter is just a sorting key for the tests, it's not directly linked to `i`.
For `Remove`, we can just go add things at the end (`offset` ensures we stay at the end of the operations list)

So this should work:
```
let mockRemove _ unused =
    let offset = match newCollection with ValueNone -> 0 | ValueSome coll -> coll.Length
    for i = 0 to unused.Count - 1 do
        operations.Add(offset + i, Remove unused.[i])
```